### PR TITLE
Try to save user data

### PR DESCRIPTION
**Pull request by khomtali**
moved and renamed files according to #12 

I tried to save user data (#9), but something went wrong and I get 500 error.
Is it right to use here PATCH request? As I understand it modifies but not creates, or regarding of 'fields' of object it doesn't matter, because it modifies it with adding new properties?

I also created menu and user profile 'page'. I found that switching between different pages in React I can do using React Router. Is it a right way for me?

### CODE REVIEW

**Comment by BTBTravis** [comment]
Hey @khomtali hmm not sure about the bug I'll check out the code and look into it. 

As for react router yeah I think it would be a great match for what you want to do with different pages! Also helpful to learn

**Comment by BTBTravis** [comment]
checking out and running your app I get an error

```
./src/components/auth/UserProvider.js
Module not found: Can't resolve '../../functions/auth' in '/Users/t.shears/play/redi/final_projects/frontends/khomtali/src/components/auth'
```

**Comment by BTBTravis** [comment]
hey @khomtali I updated your patch request. There were a few errors but it was super close. You can see changes here https://github.com/redi-js-teachers/js_sprint_2020_final_project_khomtali/pull/13/commits/f7a40f72e51a03c7985f7b4b516b5766b5cedd3e

You'll need to do a `git pull` on this feature branch before continuing working. 

The docs were correct you'll see the identifier key needed to be `id`
https://btbtravis.gitlab.io/redi-run-backend/#/user/patch_user

**Comment by khomtali** [comment]
@BTBTravis thank you for your help! I also added `try...catch` in saveUserData.js.
So, my main mistake was that I put in body not a json. sorry for other small mistakes, I tried to change something to catch the problem.

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,27 @@
 import React from 'react';
+// import { useUserContext } from './contexts/UserContext';
```
Comment (BTBTravis) [inline src/App.js:2]: shall we clean this up?

Comment (khomtali) [reply src/App.js:2]: yes, I'll delete it in the next PR.

**Pull request merged**